### PR TITLE
feat(stdlib): add pushbullet endpoint

### PR DESCRIPTION
**Pull request by affo**
Pushbullet is a service to send push notifications (https://www.pushbullet.com/).

I am using that to send notifications on my Android device.

Thought it could be interesting to add to stdlib.

### CODE REVIEW

File: stdlib/pushbullet/pushbullet.flux [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import "http"
+import "json"
+
+option defaultURL = "https://api.pushbullet.com/v2/pushes"
+
+// `pushData` sends a push notification using PushBullet's APIs.
+// `url` - string - URL of the PushBullet endpoint. Defaults to: "https://api.pushbullet.com/v2/pushes".
+// `token` - string - the api token string.  Defaults to: "".
+// `data` - object - The data to send to the endpoint. It will be encoded in JSON and sent to PushBullet's endpoint.
```
Comment (fchikwekwe) [inline stdlib/pushbullet/pushbullet.flux]: Cool idea. I'm not familiar with how we decide what notifications are useful. Are you planning to add any limitations or guidance on what data should be sent?

Comment (fchikwekwe) [reply stdlib/pushbullet/pushbullet.flux]: Like do we already have certain conventions around notifications in the web UI or something?

Comment (affo) [reply stdlib/pushbullet/pushbullet.flux]: Hey, I think this is more related to PushBullet APIs.

It's hard to find a standardized way of accessing it.

So I decided to offer a "deafult" option via the `endpoint` function.

And a `genericEndpoint` for the ones who want to specify the entire payload (auth is managed for them).

**Comment by affo** [comment]
@fchikwekwe I have looked into other packages and I saw no tests where implemented for them.

Implementing that would be mock the PushBullet API via `httptest.NewServer` and return `200` codes.

That can be done, and would verify that the script works.

On the other hand, I see the right way of doing this to write an e2e Flux test.
But, I don't think there is tooling in the framework to mock external services (at least, as far as I know).

for now,  I tested locally and it works.

File: stdlib/pushbullet/pushbullet.flux (hunk 1 above)
Comment (fchikwekwe) [reply stdlib/pushbullet/pushbullet.flux]: Ah, yeah thanks for the `slack`/`pagerduty` comparison. Helped me understand the convention a little bit better. I also guess that you're adding this as a vehicle for alerts, which makes more sense to me than a generic notification sytem.

**Comment by affo** [comment]
@fchikwekwe yeah absolutely 👍 

When there is a test framework for testing external services in Flux e2e tests, I'll  be happy to add that!

Now I have to fix things building with the Rust bits (only tried with `go build`)

File: stdlib/pushbullet/pushbullet.flux [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// For how to structure data, see https://docs.pushbullet.com/#create-push.
+// NOTE: have to default to blank data to help type system figure out that should be an object.
+pushData = (url=defaultURL, token="", data={}) => {
```
Comment (jlapacik) [inline stdlib/pushbullet/pushbullet.flux:14]: `data`'s default argument is causing a type error. The type system doesn't know how to unify an empty record `{}` with a nonempty record `{type: "note", title: title, body: text}`. Making `data` a required argument will fix this.

Comment (affo) [reply stdlib/pushbullet/pushbullet.flux:14]: I am a bit confused.

Tests pass locally but not in CI.

Tried to make `data` a required argument and I experience a failure.

🤔

**Comment by affo** [comment]
@jlapacik 

when specifying data as required I get (actually running in the REPL):

```
failed to execute query: panic: unexpected kind: got "object" expected "invalid", trace: goroutine 53 [running]:
runtime/debug.Stack(0xc000b29118, 0x44b6a65, 0xc0007f58f0)
	/usr/local/Cellar/go/1.13.8/libexec/src/runtime/debug/stack.go:24 +0x9d
github.com/influxdata/flux/values.UnexpectedKind(0xc, 0x0, 0x6359fe0, 0xc000257080)
	/Users/lorenzoaffetti/code/influxdata/flux/values/values.go:326 +0x34
github.com/influxdata/flux/values.CheckKind(0xc, 0x0)
	/Users/lorenzoaffetti/code/influxdata/flux/values/values.go:359 +0x72
github.com/influxdata/flux/compiler.(*identifierEvaluator).Eval(0xc00092d280, 0x6345340, 0xc0007012c0, 0x63583a0, 0xc0009a0aa0, 0x635a300, 0xc00092d600, 0x0, 0x0)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:532 +0xac
github.com/influxdata/flux/compiler.eval(...)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:107
github.com/influxdata/flux/compiler.(*objEvaluator).Eval(0xc00092d220, 0x6345340, 0xc0007012c0, 0x63583a0, 0xc0009a0aa0, 0x20, 0x203000, 0x203000, 0xc000b29328)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:230 +0x190
github.com/influxdata/flux/compiler.(*callEvaluator).Eval(0xc0007f5770, 0x6345340, 0xc0007012c0, 0x63583a0, 0xc0009a0aa0, 0x635a300, 0xc00092d5c0, 0x6351840, 0x6305638)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:593 +0x6c
github.com/influxdata/flux/compiler.eval(...)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:107
github.com/influxdata/flux/compiler.(*binaryEvaluator).Eval(0xc000c26cc0, 0x6345340, 0xc0007012c0, 0x63583a0, 0xc0009a0aa0, 0x635a300, 0xc00092d5c0, 0x0, 0x0)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:333 +0x66
github.com/influxdata/flux/compiler.eval(...)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:107
github.com/influxdata/flux/compiler.(*binaryEvaluator).Eval(0xc000c26d00, 0x6345340, 0xc0007012c0, 0x63583a0, 0xc0009a0aa0, 0x1, 0x1, 0x2, 0xc00092d5a0)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:337 +0xdf
github.com/influxdata/flux/compiler.eval(...)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:107
github.com/influxdata/flux/compiler.(*objEvaluator).Eval(0xc00092d1c0, 0x6345340, 0xc0007012c0, 0x63583a0, 0xc0009a0aa0, 0xc0007f5620, 0xc000c2caa8, 0x7020550, 0xc000c2ca20)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:230 +0x190
github.com/influxdata/flux/compiler.(*callEvaluator).Eval(0xc0007f57d0, 0x6345340, 0xc0007012c0, 0x63583a0, 0xc0009a0aa0, 0x6359900, 0xc00095a080, 0x0, 0x0)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:593 +0x6c
github.com/influxdata/flux/compiler.eval(...)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:107
github.com/influxdata/flux/compiler.(*objEvaluator).Eval(0xc00092d180, 0x6345340, 0xc0007012c0, 0x63583a0, 0xc0009a0aa0, 0x0, 0x0, 0x6359fe0, 0xc000257080)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:230 +0x190
github.com/influxdata/flux/compiler.eval(...)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:107
github.com/influxdata/flux/compiler.(*blockEvaluator).Eval(0xc000c26dc0, 0x6345340, 0xc0007012c0, 0x63583a0, 0xc0009a0aa0, 0xc0009a0aa0, 0x635b840, 0xc000bd00c0, 0x0)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:129 +0x9b
github.com/influxdata/flux/compiler.eval(...)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:107
github.com/influxdata/flux/compiler.compiledFn.Eval(0x6332a40, 0xc000c26dc0, 0x6351480, 0xc000c26ac0, 0x63583a0, 0xc0009a0aa0, 0x6345340, 0xc0007012c0, 0x635b840, 0xc000bd00c0, ...)
	/Users/lorenzoaffetti/code/influxdata/flux/compiler/runtime.go:71 +0xbb
github.com/influxdata/flux/execute.(*rowFn).eval(0xc000bd0120, 0x6345340, 0xc0007012c0, 0x0, 0x6357520, 0xc0009a6140, 0x0, 0x190, 0x190, 0xc000303040, ...)
	/Users/lorenzoaffetti/code/influxdata/flux/execute/row_fn.go:173 +0x2ae
github.com/influxdata/flux/execute.(*RowMapFn).Eval(0xc000bd0120, 0x6345340, 0xc0007012c0, 0x0, 0x6357520, 0xc0009a6140, 0xc000b29a80, 0x401026e, 0x540e6e0, 0xc000303048)
	/Users/lorenzoaffetti/code/influxdata/flux/execute/row_fn.go:253 +0x75
github.com/influxdata/flux/stdlib/universe.(*mapTransformation).Process.func1(0x6357520, 0xc0009a6140, 0xc000b29c00, 0x400e818)
	/Users/lorenzoaffetti/code/influxdata/flux/stdlib/universe/map.go:173 +0x2cc
github.com/influxdata/flux/execute.(*ColListTable).Do(0xc0009a6140, 0xc000c26e00, 0xc0000dea50, 0x2)
	/Users/lorenzoaffetti/code/influxdata/flux/execute/table.go:1409 +0x78
github.com/influxdata/flux/stdlib/universe.(*mapTransformation).Process(0xc00047e8c0, 0xcc531e8be529b31c, 0xbebd4d04a5dec09d, 0x6348600, 0xc0009a6140, 0x1, 0x0)
	/Users/lorenzoaffetti/code/influxdata/flux/stdlib/universe/map.go:170 +0x135
github.com/influxdata/flux/execute.processMessage(0x6345340, 0xc0007012c0, 0x634a0c0, 0xc00047e8c0, 0x6333080, 0xc00095a000, 0xc000c5de9a, 0x0, 0x0)
	/Users/lorenzoaffetti/code/influxdata/flux/execute/transport.go:206 +0x2d6
github.com/influxdata/flux/execute.(*consecutiveTransport).processMessages(0xc000bd0180, 0x6345340, 0xc0007012c0, 0xa)
	/Users/lorenzoaffetti/code/influxdata/flux/execute/transport.go:159 +0xc6
github.com/influxdata/flux/execute.(*poolDispatcher).run(0xc000830d80, 0x6345340, 0xc0007012c0)
	/Users/lorenzoaffetti/code/influxdata/flux/execute/dispatcher.go:132 +0x65
github.com/influxdata/flux/execute.(*poolDispatcher).Start.func1(0xc000830d80, 0x6345340, 0xc0007012c0)
	/Users/lorenzoaffetti/code/influxdata/flux/execute/dispatcher.go:86 +0xcd
created by github.com/influxdata/flux/execute.(*poolDispatcher).Start
	/Users/lorenzoaffetti/code/influxdata/flux/execute/dispatcher.go:63 +0x7e
```

Should happen in the compiler at runtime, so no Rust involved.

**Comment by affo** [comment]
@jlapacik commented out `genericEndpoint`, left `pushData` because it still works by itself 👍

**Pull request merged**